### PR TITLE
[TEP-0104]: Descope to runtime only

### PR DESCRIPTION
**Pull request by lbernick**
This commit updates TEP-0104 to specify that we will implement
Task-level resource requirements for TaskRuns only to start,
and we may consider adding support on Tasks in the future.

/kind tep

### CODE REVIEW

**Comment by lbernick** [comment]
@dibyom @jerop @vdemeester @leeonlee @austinzhao-go PTAL

File: teps/0104-tasklevel-resource-requirements.md [hunk 1]
```diff
@@ -103,18 +103,15 @@ depending on its inputs. In addition, LimitRanges don’t distinguish between Te
...
 metadata:
-  name: image-build-task
+  name: image-build-taskrun
 spec:
-  steps:
-  - name: step-1
-  - name: step-2
   resources:
     requests:
       memory: 1Gi
```
Comment (austinzhao-go) [inline teps/0104-tasklevel-resource-requirements.md:117]: thinking here we perhaps still will meet a "field overloading" problem for `TaskRun`:

```
type TaskRunSpec struct {
	// +optional
	Debug *TaskRunDebug `json:"debug,omitempty"`
	// +optional
	// +listType=atomic
	Params []Param `json:"params,omitempty"`
	// +optional
	Resources *TaskRunResources `json:"resources,omitempty"`
```
https://github.com/austinzhao-go/pipeline/blob/tep-0104-per-task-resources-requirements/pkg/apis/pipeline/v1beta1/taskrun_types.go#L44

Comment (lbernick) [reply teps/0104-tasklevel-resource-requirements.md:117]: You're completely right -- addressed in https://github.com/tektoncd/community/pull/726

File: teps/0104-tasklevel-resource-requirements.md [hunk 2]
```diff
@@ -262,9 +291,18 @@ spec:
...
+metadata:
+  name: my-task
+spec:
+  taskRef:
+    name: my-task
   computeResources:
-    limits:
-      cpu: 2
+    requests:
+      cpu: 1.5
```
Comment (austinzhao-go) [inline teps/0104-tasklevel-resource-requirements.md:305]: here could still be `limits`

**Review by austinzhao-go — CHANGES_REQUESTED** [review]

**Review by jerop — APPROVED** [review]

File: teps/0104-tasklevel-resource-requirements.md [hunk 3]
```diff
@@ -125,17 +122,17 @@ spec:
 
 This field should also be added to [PipelineRun.TaskRunSpecs][taskRunSpecs].
```
Comment (austinzhao-go) [inline teps/0104-tasklevel-resource-requirements.md:123]: think we still need this field under `PipelineRun` right? 

there is no "legacy" `Resources` field, but will keep using `ComputeResources` for a consistent naming

Comment (lbernick) [reply teps/0104-tasklevel-resource-requirements.md:123]: yes, it still makes sense to have this in PipelineRun.Spec.TaskRunSpecs under "computeResources"

File: teps/0104-tasklevel-resource-requirements.md (hunk 2 above)
Comment (lbernick) [reply teps/0104-tasklevel-resource-requirements.md]: thanks austin, this was a merge conflict error-- fixed now

**Comment by jerop** [comment]
/lgtm 

approved by both assignees - @afrittoli and @jerop - so merging offline

**Pull request merged**